src/data/set/prod.lean

```diff
@@ -40,10 +40,18 @@ lemma mem_prod_eq {p : α × β} : p ∈ s ×ˢ t = (p.1 ∈ s ∧ p.2 ∈ t) :=
 
 @[simp] lemma mem_prod {p : α × β} : p ∈ s ×ˢ t ↔ p.1 ∈ s ∧ p.2 ∈ t := iff.rfl
 
+@[simp] lemma not_mem_prod {p : α × β} : p ∉ s ×ˢ t ↔ p.1 ∉ s ∨ p.2 ∉ t := not_and_distrib
+
 @[simp] lemma prod_mk_mem_set_prod_eq : (a, b) ∈ s ×ˢ t = (a ∈ s ∧ b ∈ t) := rfl
 
 lemma mk_mem_prod (ha : a ∈ s) (hb : b ∈ t) : (a, b) ∈ s ×ˢ t := ⟨ha, hb⟩
 
+instance decidable_mem_prod [hs : decidable_pred s] [ht : decidable_pred t] (x : α × β) :
+  decidable (x ∈ s ×ˢ t) :=
+decidable.rec_on (hs x.1) (λ ha, is_false (set.not_mem_prod.2 $ or.inl ha))
+  (λ ha, decidable.rec_on (ht x.2) (λ hb, is_false (set.not_mem_prod.2 $ or.inr hb))
+    (λ hb, is_true $ set.mem_prod.2 ⟨ha, hb⟩))
```

Comment:
It doesn't make sense to say that a `set` is decidable; it's _membership_ of the set that's decidable:
```suggestion
instance decidable_mem_prod [decidable_pred (∈ s)] [decidable_pred (∈ t)] :
  decidable_pred (∈ s ×ˢ t) :=
λ _, and.decidable
```

Reply:
Sure, that makes sense. I've also changed the existing instance for `diagonal` to use this notation instead.